include/deal.II/lac/sparse_matrix_ez.h

```diff
@@ -683,6 +683,17 @@ class SparseMatrixEZ : public Subscriptor
   const_iterator begin (const size_type r) const;
 
   /**
+   * Iterator on diagonal entry of row @p r.
+   *
+   * @warning If the current row has no diagonal element, then the returned
+   * iterator will be same as end(r), i.e., an iterator points to first element
+   * of next row or same place behind end of the entire matrix.
+   * So, do not dereference or query information to the returned iterator blindly
+   * unless you can make sure your matrix do have a diagonal element in row @p r.
+   */
+  const_iterator diagonal (const size_type r) const;
```

Comment:
I think this is not a good name. It suggests that you get the *entire* diagonal, not just one element, and it says nothing about the fact that you get an iterator. In fact, since you return a `const_iterator` with which one can not do anything that just read from the element, why not just return the value of the diagonal element? (For which I think there is a function already.)